src/parser.js

```diff
@@ -339,12 +335,16 @@ define([
             mug = form.getMugByPath(path);
         }
         mug = adapt(mug, form);
-        mug.parentMug = parentMug;
+        var node = form.tree.getNodeFromMug(mug);
+        if (node && node.parent.value !== parentMug) {
+            mug.p.dataParent = node.parent.getAbsolutePath();
+            form.tree.insertMug(mug, 'into', parentMug);
```

Comment:
Hmm, this could be problematic. Will the control tree order be preserved in all cases? May need to add a test or two where the special mug with a dataParent comes before, after, and/or sandwiched between other normal mugs.